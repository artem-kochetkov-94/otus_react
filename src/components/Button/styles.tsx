import styled from '@emotion/styled'
import { css } from '@emotion/core'
import { ButtonProps } from './'

const ButtonBase = css`
  display: inline-flex;
  justify-content: center;
  min-width: 200px;
  padding: 5px 15px;
  font-size: 16px;
  color: #000;
`

const ButtonPrimary = css`
  background-color: blue;
  color: #fff;
`

const ButtonDangerous = css`
  background-color: red;
  color: #fff;
`

const ButtonDisabled = css`
  opacity: 0.2;
`

export const ButtonStyled = styled.button`
  ${ButtonBase};
  ${({ disabled }: ButtonProps) => (disabled ? ButtonDisabled : '')};
  ${({ customType }: ButtonProps) => {
    switch (customType) {
      case 'primary':
        return ButtonPrimary
      case 'dangerous':
        return ButtonDangerous
      default:
        return ''
    }
  }}
`
