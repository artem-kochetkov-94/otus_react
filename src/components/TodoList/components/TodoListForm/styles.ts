import styled from '@emotion/styled'

export const Input = styled.input`
  display: inline-flex;
  min-width: 200px;
  padding: 5px;
`

export const FormField = styled.div`
  margin-bottom: 20px;
`

export const Form = styled.form`
  padding: 10px;
  border: 1px solid #000;
`

export const Label = styled.p`
  margin-bottom: 5px;
`
