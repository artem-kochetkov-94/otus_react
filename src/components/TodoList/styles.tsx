import styled from '@emotion/styled'
import { css } from '@emotion/core'

const ButtonPageBase = css`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 30px;
  height: 30px;
  margin-right: 10px;
  margin-bottom: 10px;
  border: 1px solid #000;
  font-size: 14px;
  color: #000;
`

const ButtonPageActive = css`
  background-color: blue;
  color: white;
`

const PaginationBase = css`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 10px;
`

export const ButtonPage = styled.div`
  ${ButtonPageBase};
  ${({ isActive }) => (isActive ? ButtonPageActive : '')};
`

export const Pagination = styled.div`
  ${PaginationBase}
`
