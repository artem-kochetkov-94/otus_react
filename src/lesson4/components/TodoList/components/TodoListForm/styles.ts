import styled from '@emotion/styled'

export const Input = styled.input`
  display: inline-flex;
  min-width: 200px;
  padding: 5px;
`
