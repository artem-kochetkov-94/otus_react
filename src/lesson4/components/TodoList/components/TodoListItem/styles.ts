import styled from '@emotion/styled'

export const Wrapper = styled.div`
  margin-bottom: 20px;
  padding: 10px;
  border: 1px solid #000;
`

export const TaskComplete = styled.div`
  margin-bottom: 5px;
  font-size: 16px;
  font-weight: 700;
  color: green;
`
