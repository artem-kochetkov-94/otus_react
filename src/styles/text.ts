import styled from '@emotion/styled'

export const Text = styled.p`
  margin: 0;
  margin-bottom: 10px;
  font-weight: 700;
  font-size: 18px;
`
