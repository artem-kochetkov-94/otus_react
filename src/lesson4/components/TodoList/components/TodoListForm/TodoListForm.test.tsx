import React from 'react'
import { mount } from 'enzyme'
import { TodoListForm } from './'
import renderer from 'react-test-renderer'
import { Form } from './styles'

describe('Default render tests', () => {
  const onClick = jest.fn()

  it('renders snapshot', () => {
    expect(
      renderer.create(<TodoListForm onCreate={onClick} />).toJSON(),
    ).toMatchSnapshot()
  })
})

describe('Actions tests', () => {
  it('The button is locked if the field is empty', () => {
    const onClick = jest.fn()

    const todoListFormWrapper = mount(<TodoListForm onCreate={onClick} />)

    expect(
      todoListFormWrapper.find('button.add-task').props().disabled,
    ).toEqual(true)

    todoListFormWrapper.find('button.add-task').simulate('click')
    expect(onClick).not.toHaveBeenCalled()
  })

  it('Create task', () => {
    const onClick = jest.fn()
    const fakeEvent = { preventDefault: () => console.log('preventDefault') }

    const todoListFormWrapper = mount(<TodoListForm onCreate={onClick} />)
    todoListFormWrapper
      .find('input[name="task-header"]')
      .simulate('change', { target: { value: 'test' } })
    // todoListFormWrapper.find('button.add-task').simulate('click')

    todoListFormWrapper.find(Form).simulate('submit', fakeEvent)

    expect(onClick).toHaveBeenCalled()
    expect(
      todoListFormWrapper.find('input[name="task-header"]').prop('value'),
    ).toEqual('')
  })
})
