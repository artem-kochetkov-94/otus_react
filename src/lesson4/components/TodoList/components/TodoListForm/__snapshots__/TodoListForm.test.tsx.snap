// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render tests renders snapshot 1`] = `
.emotion-0 {
  margin: 0;
  margin-bottom: 10px;
  font-weight: 700;
  font-size: 18px;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 200px;
  padding: 5px;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 200px;
  padding: 5px 15px;
  font-size: 16px;
  color: #000;
  opacity: 0.2;
  background-color: blue;
  color: #fff;
}

<div
  className="todo-list-form"
>
  <p
    className="emotion-0"
  >
    Создать задачу
  </p>
  <input
    className="emotion-1"
    onChange={[Function]}
    value=""
  />
  <button
    className="add-task emotion-2"
    disabled={true}
    onClick={[Function]}
  >
    Добавить
  </button>
</div>
`;
